geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/QueryFilterSplitter.scala

```diff
@@ -319,10 +397,26 @@ class QueryFilterSplitter(sft: SimpleFeatureType) extends LazyLogging {
   }
 
   /**
-   * Will perform a full table scan - used when we don't have anything better. Uses record table
+   * Will perform a full table scan - used when we don't have anything better. Uses z2 table
    */
-  private def fullTableScanOption(filter: Option[Filter]): FilterPlan =
-    FilterPlan(Seq(QueryFilter(StrategyType.RECORD, Seq(Filter.INCLUDE), filter)))
+  private def fullTableScanOption(filter: Option[Filter]): FilterPlan = {
+    if (supported.contains(Z2Table)) {
+      FilterPlan(Seq(QueryFilter(StrategyType.Z2, Seq(Filter.INCLUDE), filter)))
+    } else if (supported.contains(RecordTable)) {
+      FilterPlan(Seq(QueryFilter(StrategyType.RECORD, Seq(Filter.INCLUDE), filter)))
+    } else {
```

Comment:
can we also fall back on the z3 table if we were running a schema with just z3 ?